c10/test/core/SymInt_test.cpp

```diff
@@ -21,4 +22,32 @@ TEST(SymIntTest, ConcreteInts) {
 TEST(SymIntTest, CheckRange) {
   EXPECT_FALSE(SymInt::check_range(INT64_MIN));
 }
+
+TEST(SymIntTest, SingletonSymNode) {
+  auto a = c10::SymInt(
+      c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(1)));
+  auto b = c10::SymInt(
+      c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(1)));
+  auto c = c10::SymInt(
+      c10::SymNode(c10::make_intrusive<c10::SingletonSymNodeImpl>(2)));
+  auto d = c10::SymInt(3);
+
+  ASSERT_TRUE(a == a);
+  ASSERT_TRUE(a == b);
+  ASSERT_FALSE(a != a);
+  ASSERT_FALSE(a != b);
+  ASSERT_FALSE(a == c);
+  ASSERT_TRUE(a != c);
+
+  // Tentaively throw an error when comparing with a non-singleton, this is not
+  // necessarily the right behavior.
```

Comment:
Yeah, I could see an argument that comparison with integers just returns False. We can change this when we get more experience with it in the frontend.